browser/brave_rewards/rewards_service_factory.cc

```diff
@@ -90,8 +92,7 @@ RewardsServiceFactory::RewardsServiceFactory()
 #endif
 
 #if BUILDFLAG(BRAVE_REWARDS_ENABLED) && !defined(OS_ANDROID)
-  registrar_.Add(this, chrome::NOTIFICATION_PROFILE_CREATED,
-                 content::NotificationService::AllSources());
+  g_browser_process->profile_manager()->AddObserver(this);
```

Comment:
missing RemoveObserver which could cause a crash

Reply:
Oops! Good catch, fixed in https://github.com/brave/brave-core/pull/7866/commits/cd715dcdb93d3730ff06b22afef9e3cc80ae6c10

Reply:
Actually had to ammend https://github.com/brave/brave-core/commit/cd715dcdb93d3730ff06b22afef9e3cc80ae6c10 and protect the call or a crash would be observed in the `TorProfileManagerExtensionTest.SwitchToTorProfileIncognitoEnabled`, where the `g_browser_process` is no longer valid by the time the RewardsServiceFactory object gets destroyed.

New fix here: https://github.com/brave/brave-core/commit/fd2eb16fd863cd82da2fac9247a0b2bf7dd1a791